pubsub/driver/driver.go

```diff
@@ -0,0 +1,46 @@
+package driver
+
+import "context"
+
+type AckID interface{}
+
+type Message struct {
+	// Body contains the content of the message.
+	Body []byte
+
+	// Attributes has key/value metadata for the message.
+	Attributes map[string]string
+
+	// AckID identifies the message on the server. It can be used to ack
```

Comment:
Since this is the driver interface, I suggest focusing the documentation on someone who is going to implement it. There is only one user (the concrete type) so (unusually) the user is less important.

So for example, on this one, something like:

// AckID should be set to something identifying the message on the server. It may be passed to SendAcks() to ack the message.